t1/src/decoder/Decoder.scala

```diff
@@ -31,7 +33,186 @@ trait BoolField extends BoolDecodeField[Op] with FieldName {
   def genTable(op: Op): BitPat = if (dontCareCase(op)) dc else if (value(op) && (containsLSU || op.notLSU)) y else n
 }
 
+case class SpecialAux(name: String, vs: Int, value: String)
+case class SpecialMap(name: String, vs: Int, data: Map[String, String])
+case class SpecialAuxInstr(instrName: String, vs: Int, value: String, name: String)
+case class Op(tpe: String, funct6: String, funct3: String,
+              name: String, special: Option[SpecialAux], notLSU: Boolean) extends DecodePattern {
+  val funct3Map: Map[String, String] = Map(
+    "IV" -> "000",
+    "IX" -> "100",
+    "II" -> "011",
+    "MV" -> "010",
+    "MX" -> "110",
+    "FV" -> "001",
+    "FF" -> "101"
```

Comment:
OPCFG is missed, and please add documentation for these functions, give reference link and **align** to specification:
e.g.
```suggestion
    // Refer to [https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#101-vector-arithmetic-instruction-encoding].
    "OPIVV" -> "000",
    "OPFVV" -> "001",
    "OPMVV" -> "010",
    "OPIVI" -> "011",
    "OPIVX" -> "100",
    "OPFVF" -> "101",
    "OPMVX" -> "110",
    "OPCFG" -> "111",
```